test/integration/sample_test.go

```diff
@@ -58,20 +73,35 @@ var skipDiscoverDirs = map[string]bool{
 	".git":  true,
 }
 
-// discoverTestCases discovers individual sample directories in the parent directory.
+// discoverTestCaseGroups discovers individual sample directories in the parent directory
+// and assigns them to a test group based on projects provided.
 // It also skips known directories that are not samples.
-func discoverTestCases(t *testing.T) []string {
+func discoverTestCaseGroups(t *testing.T, projects []string) []*testGroups {
 	t.Helper()
-	tc := []string{}
-	samples, err := ioutil.ReadDir(sampleDir)
+	samples := []string{}
+	dirs, err := ioutil.ReadDir(sampleDir)
 	if err != nil {
 		t.Fatal(err)
 	}
-	for _, f := range samples {
+	for _, f := range dirs {
 		if !f.IsDir() || skipDiscoverDirs[f.Name()] {
 			continue
 		}
-		tc = append(tc, f.Name())
+		samples = append(samples, f.Name())
+	}
+	sort.Strings(samples)
+
+	// One test group is associated to one project.
+	groups := []*testGroups{}
+	for i, project := range projects {
+		groups = append(groups, &testGroups{projectID: project, samples: []string{}, group: i})
+	}
+	// Rather than chunking we assign them in a round robin fashion as some samples like sql takes more time.
+	// Chunking would result in all sql* assigned to a single project.
+	// We sort the sample slice beforehand so assignments should be stable for a given run.
+	for i, sample := range samples {
+		idx := i % len(projects)
```

Comment:
nit: rename to groupIndex since i & idx are similar

Reply:
done